pkg/controller/dns/external_dns_crd.go

```diff
@@ -0,0 +1,118 @@
+package dns
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/Azure/aks-app-routing-operator/api/v1alpha1"
+	"github.com/Azure/aks-app-routing-operator/pkg/config"
+	"github.com/Azure/aks-app-routing-operator/pkg/controller/controllername"
+	"github.com/Azure/aks-app-routing-operator/pkg/controller/keyvault"
+	"github.com/Azure/aks-app-routing-operator/pkg/controller/metrics"
+	"github.com/Azure/aks-app-routing-operator/pkg/manifests"
+	"github.com/Azure/aks-app-routing-operator/pkg/util"
+	"github.com/go-logr/logr"
+	"github.com/hashicorp/go-multierror"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var ExternalDNSCRDControllerName = controllername.New("externaldns", "crd")
+
+type ExternalDNSCRDController struct {
+	config *config.Config
+	client client.Client
+	events record.EventRecorder
+}
+
+func NewExternalDNSCRDController(manager ctrl.Manager, config config.Config) error {
+	return ExternalDNSCRDControllerName.AddToController(ctrl.NewControllerManagedBy(manager).
+		For(&v1alpha1.ExternalDNS{}).
+		Owns(&appsv1.Deployment{}).
+		Owns(&corev1.ConfigMap{}), manager.GetLogger()).
+		Complete(&ExternalDNSCRDController{
+			config: &config,
+			client: manager.GetClient(),
+			events: manager.GetEventRecorderFor("aks-app-routing-operator"),
+		})
+}
+
+func (e *ExternalDNSCRDController) Reconcile(ctx context.Context, req ctrl.Request) (res ctrl.Result, err error) {
+	defer func() {
+		metrics.HandleControllerReconcileMetrics(ExternalDNSCRDControllerName, res, err)
+	}()
+
+	// set up logger
+	logger, err := logr.FromContext(ctx)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("creating logger: %w", err)
+	}
+	logger = ExternalDNSCRDControllerName.AddToLogger(logger).WithValues("namespace", req.Namespace, "name", req.Name)
+
+	obj := &v1alpha1.ExternalDNS{}
+	if err = e.client.Get(ctx, req.NamespacedName, obj); err != nil {
+		logger.Error(client.IgnoreNotFound(err), "failed to get externaldns object, will ignore not found error")
```

Comment:
i like not logging at all if it's not found

Reply:
so I was thinking about this - should I just add extra lines to log if it's not a not-found error? I wouldn't want to end up in a situation where we can't pull our CRDs (for some weird reason) and we don't have immediate logging telling us why

Reply:
yes. What's the problem with just logging if it's not not found?

Reply:
It's extremely common for not found to be the case because controller runtime requeues on delete

Reply:
ok yeah sounds good I can have this operate in the same way our other ones do, I'll have it log if not found otherwise log/return error